Helper/OmiseHelper.php

```diff
@@ -370,12 +370,9 @@ public function isCreditCardPaymentMethod($paymentMethod)
     /**
      * Validate whether the a URI was triggered by Omise server or not
      */
-    public function validate3DSReferer()
+    public function isUserOriginated()
     {
-        $refererValue = $this->header->getHttpReferer();
-        $isProduction = strpos($refererValue, 'https://api.omise.co') === 0;
-        $isStaging = strpos($refererValue, 'https://api.staging-omise.co') === 0;
-
-        return $isProduction || $isStaging;
+        // "none" means the request is a user-originated operation
+        return 'none' === $_SERVER['HTTP_SEC_FETCH_SITE'];
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Sec-Fetch-Site there is no support of this on Safari. should we have a fallback as well? please give it a test on the safari web browser

Reply:
Thanks. I will look into it.

Reply:
@kan98 I have re-added the logic that relies on`referer` for Safari. Even though it's less reliable, it's better than having nothing. I couldn't find other solution that would meet our objective.